### DISCUSSION TITLE
Run only one DAG at a time

### DISCUSSION THREAD

**Discussion by nitra-1** [apache/airflow#29664] [General]
Is there any setting which can enable execution of only one DAG at a time, 
At the same time if some other DAG is overlapped with the time of the execution of the current running DAG then it will not start or get queued.

**Comment by potiuk** [comment]
Not really. I am not aware a way you can have an exclusive One DAG at a time setting. Scheduler is really "Task" oriented not DAG oriented in this case.

One thing you might try thoguh is to introduce dependencies between DAGS via https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html - you could "chain" your DAGs, so that there are dependencies between them  - this is not exactly what you want especially if your DAGs are running with different schedule (one daily, one hourly etc.), but if all the DAGs you want to limit from overlapping run roughly at the same schedule (all of them dailiy in the night for example). Then you could have the DAGs written in the way that one DAG would produce a dataset when it finishes, and another consumes it (and finishing DAG 1 would trigger DAG 2).

When it come to limting specfic tasks from running in parallel, there are multiple options in configuration, pools, queues, task parameters, etc.
